cookbook/upgrade/deprecation_warnings.rst

```diff
@@ -0,0 +1,56 @@
+What do these "XXX is deprecated " E_USER_DEPRECATED Warnings mean?
+===================================================================
+
+Starting in Symfony 2.7, if you use a deprecated class, function or option,
+Symfony triggers an ``E_USER_DEPRECATED`` error. Internally, that looks something
+like this::
+
+    trigger_error(
+        'The fooABC method is deprecated since version 2.4 and will be removed in 3.0.',
+        E_USER_DEPRECATED
+    );
+
+This is great, because you can check your logs to know what needs to change
+before you upgrade. In the Symfony Framework, the number of deprecated calls
+shows up in the web debug toolbar. And if you install the `phpunit-bridge`_,
+you can get a report of deprecated calls after running your tests.
+
+How can I Silence the Warnings?
+-------------------------------
+
+As useful as these are, you don't want them to show up while developing and
+you may also want to silence them on production to avoid filling up your
+error logs. To do that, add ``~E_USER_DEPRECATED`` to your ``error_reporting``
+setting in ``php.ini``:
+
+.. code-block:: ini
+
+    ; before
+    error_reporting = E_ALL
+    ; after
+    error_reporting = E_ALL & ~E_USER_DEPRECATED
+
+Alternatively, you can set this directly in bootstrap of your project::
```

Comment:
I agree with telling a specific file, but in the framework, this is taken care of for you automatically (actually, I need to add a note about that - similar to https://github.com/symfony/symfony/pull/14776/files#diff-722b43ae068b297ed9d04d3b08ec16cfR14.

And of course, outside of the framework, we don't know what files they'll have. Any other suggestion? And I'm not sure about the performance hit of calling this one function. After all, we *are* actually doing this in the framework.